python/tach/core/config.py

```diff
@@ -128,7 +137,7 @@ def add_dependency_to_module(self, module: str, dependency: str):
             self.modules.append(ModuleConfig(path=module, depends_on=[dependency]))
         else:
             # Config already exists, set the union of existing dependencies and new ones
-            new_dependencies = set(current_module_config.depends_on) | {dependency}
+            new_dependencies = set(current_module_config.depends_on) | {dependency}  # pyright: ignore[reportUnhashable]
```

Comment:
Is Dependency actually hashable but pyright can't detect it?

Reply:
Yep! Because of `model_config = {"frozen": True}`See https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.frozen